src/components/ViewerComponent.vue

```diff
@@ -33,7 +33,7 @@
 export default {
 	name: 'ViewerComponent',
 	components: {
-		EditorWrapper: () => import(/* webpackChunkName: "editor" */'./EditorWrapper'),
+		EditorWrapper: () => import(/* webpackChunkName: "editor" */'@nextcloud/text-editor'),
```

Comment:
We can move it to a local file and re-export it as a local component.
It will turn easier to adjust lazy load and other possible adjustments.

```js
const EditorWrapper = () => import(/* webpackChunkName: "editor" */'@nextcloud/text-editor')

export default EditorWrapper
```

Reply:
Also, we can access any exported value inside `@nextcloud/text-editor`.

```js
const EditorWrapper = () => import(/* webpackChunkName: "editor" */'@nextcloud/text-editor')
  .then(package => package.default)
```

It is important because we are using named exports like `store`

```js
import { store } from '@nextcloud/text-editor'
```

Lazy loading the default export and do not use lazy loading in named exports are incompatibles.

To make it, we must use lazy loading in named imports too.

```js
const Store = () => import(/* webpackChunkName: "editor" */'@nextcloud/text-editor')
  .then(package => package.store)
```

But it does not make a lot of sense.
In our architecture, we can't use the async store without changing some parts of code and flow.

To make it possible we need to split the store and components into different parts/files of the `@nextcloud/text-editor` package.
It is possible, but I don't know if webpack will handle it well.

Reply:
Well... it's possible - the store for example already is in a file of its own and importing just that file worked fine. I also noticed that highlight.js also uses paths into the package as more fine grained imports. So that seems to be an somewhat established option. Not pretty but maybe enough to get a first working iteration.

I might just use the file paths into the alias in src and then index.js to define the exports of the package that we can then use in collectives. I think there we just want the tiptap extensions for displaying the tables etc. nicely.

We might also want the editor wrapper to get rid of the magic we use to extract the editor component from the viewer. But that could be a separate step or we could just reference it by path to avoid bloating the read only editor in collectives if we include the `EditorWrapper` in index.js.